server/repository/src/migrations/v1_07_00/mod.rs

```diff
@@ -31,6 +32,18 @@ mod stocktake_line_add_item_link_id;
 mod item_link_create_table;
 pub(crate) struct V1_07_00;
 
+impl Migration for V1_07_00 {
```

Comment:
I had meant to commit this separately because it's a preference of mine that the effective entry point of the code of a file is at the top rather than in the middle of the file, but someone might argue for top down declarations of fns cause tradition or something, to which I say we're using rust, not C, so we can do nice things 😆.

Reply:
don't really mind much but find it nice to have everything defined before it is used but yes this can push the main fn quite a bit down.

Reply:
Reminds me of when I've made huge comments before and put the "TL;DR:" at the bottom rather than the top - feedback I've received is put "TL;DR:" , summary or otherwise at the top so people can see it before reading a wall of text lol. Other example is abstracts for scientific papers are at the start.

Still, totally personal preference and many languages like C _require_ declarations before use.

Reply:
Just played with VSCode to jump to the main fn. The file outline uses too much space for my tast to keep it open all the time, but there is this snake-bar at the top that seems to be useful:

![image](https://github.com/msupply-foundation/open-msupply/assets/88299195/8222f199-a8f9-47b9-bc77-8b7d9f2e6d7b)